ethereum/contracts/query/QueryResponse.sol

```diff
@@ -0,0 +1,246 @@
+            (r.responses[idx].request, reqIdx) = response.sliceUnchecked(reqIdx, len);
+
+            (len, respIdx) = response.asUint32Unchecked(respIdx);
+            (r.responses[idx].response, respIdx) = response.sliceUnchecked(respIdx, len);
+
+            unchecked { ++idx; }
+        }
+
+        checkLength(response, respIdx);
+        return r;
+    }
+
+    /// @dev parseEthCallQueryResponse parses a ParsedPerChainQueryResponse for an ETH call per-chain query.
+    function parseEthCallQueryResponse(ParsedPerChainQueryResponse memory pcr) public pure returns (EthCallQueryResponse memory r) {
+        if (pcr.queryType != QT_ETH_CALL) {
+                revert UnsupportedQueryType();
+        }
+
+        uint reqIdx = 0;
+        uint respIdx = 0;
+
+        uint32 len;
+        (len, reqIdx) = pcr.request.asUint32Unchecked(reqIdx); // block_id_len
+
+        (r.requestBlockId, reqIdx) = pcr.request.sliceUnchecked(reqIdx, len);
+
+        uint8 numBatchCallData;
+        (numBatchCallData, reqIdx) = pcr.request.asUint8Unchecked(reqIdx);
+
+        (r.blockNum, respIdx) = pcr.response.asUint64Unchecked(respIdx);
+
+        (r.blockHash, respIdx) = pcr.response.asBytes32Unchecked(respIdx);
+
+        (r.blockTime, respIdx) = pcr.response.asUint64Unchecked(respIdx);
+
+        uint8 respNumResults;
+        (respNumResults, respIdx) = pcr.response.asUint8Unchecked(respIdx);
+        if (respNumResults != numBatchCallData) {
+                revert UnexpectedNumberOfResults();
+        }
+
+        r.result = new EthCallData[](numBatchCallData);
+
+        // Walk through the call data and results in lock step.
+        for (uint idx = 0; idx < numBatchCallData;) {
+            (r.result[idx].contractAddress, reqIdx) = pcr.request.asAddressUnchecked(reqIdx);
+
+            (len, reqIdx) = pcr.request.asUint32Unchecked(reqIdx); // call_data_len
+            (r.result[idx].callData, reqIdx) = pcr.request.sliceUnchecked(reqIdx, len);
+
+            (len, respIdx) = pcr.response.asUint32Unchecked(respIdx); // result_len
+            (r.result[idx].result, respIdx) = pcr.response.sliceUnchecked(respIdx, len);
+
+            unchecked { ++idx; }
+        }
+
+        checkLength(pcr.request, reqIdx);
+        checkLength(pcr.response, respIdx);
+        return r;
+    }
+
+    /**
+     * @dev verifyQueryResponseSignatures verifies the signatures on a query response. It calls into the Wormhole contract.
+     * IWormhole.Signature expects the last byte to be bumped by 27 
+     * see https://github.com/wormhole-foundation/wormhole/blob/637b1ee657de7de05f783cbb2078dd7d8bfda4d0/ethereum/contracts/Messages.sol#L174
+     */
+    function verifyQueryResponseSignatures(address _wormhole, bytes memory response, IWormhole.Signature[] memory signatures) public view {
+        IWormhole wormhole = IWormhole(_wormhole);
+        // TODO: make a verifyCurrentQuorum call on the core bridge so that there is only 1 cross call instead of 4
```

Comment:
Stale comment.

Reply:
Updated the comment to say this change might be useful some day.